### PR TITLE
Add `Assignments Remaining` column to Unassigned Table

### PR DESCRIPTION
**Pull request by mushtaqak**
The purpose of this PR is to add a new column `Assignments Remaining` to Unassigned Table that will display remaining assignments for each voucher code.

[ENT-1706](https://openedx.atlassian.net/browse/ENT-1706)

### CODE REVIEW

**Comment by mushtaqak** [comment]
@adamstankiewicz Can you please look at this?

File: src/components/CouponDetails/index.jsx [hunk 1]
```diff
@@ -403,6 +407,18 @@ class CouponDetails extends React.Component {
...
     }
 
+    // `assignments_remaining` column
+    if (value === 'unassigned' && getColumnIndexForKey('assignments_remaining') === -1) {
+      // Add `assignments_remaining` column if it doesn't already exist.
+      tableColumns.splice(3, 0, {
+        label: 'Assignments Remaining',
+        key: 'assignments_remaining',
+      });
+    } else if (value !== 'unassigned' && getColumnIndexForKey('assigned_to') > -1) {
```
Comment (adamstankiewicz) [inline src/components/CouponDetails/index.jsx]: Are you meaning to use `assigned_to` here versus `assignments_remaining`?

Comment (mushtaqak) [reply src/components/CouponDetails/index.jsx]: Oh, yeah. Nice catch ;)

Comment (adamstankiewicz) [reply src/components/CouponDetails/index.jsx]: Once that's fixed up, I'll give a 👍!

**Pull request merged**